### PR TITLE
Improve aeron recipe

### PR DESCRIPTION
**Pull request by uilianries**
Signed-off-by: Uilian Ries <uilianries@gmail.com>Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/aeron/all/conanfile.py [hunk 1]
```diff
@@ -107,48 +87,43 @@ def package(self):
...
-        self.user_info.RESOURCES_DIR = resources_folder
-        self.user_info.VERSION = self.version
+        self.cpp_info.includedirs.append(os.path.join("include", "aeron"))
+        if self.settings.compiler == "Visual Studio":
+            self.cpp_info.defines.append("_ENABLE_EXTENDED_ALIGNED_STORAGE")
+
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["m", "pthread"]
+        elif self.settings.os == "Windows":
+            self.cpp_info.system_libs = ["wsock32", "ws2_32", "Iphlpapi"]
```
Comment (lukaszlaszko) [inline recipes/aeron/all/conanfile.py:128]: I think those should be added only when build_aeron_driver = True

File: recipes/aeron/all/conanfile.py [hunk 2]
```diff
@@ -107,48 +87,43 @@ def package(self):
...
-        files = os.listdir(old_include_folder)
         with tools.chdir(self.package_folder):
-            for f in files:
-                shutil.move(os.path.join(old_include_folder, f), new_include_folder)
+            tools.rename("include", "old_include")
+            os.makedirs("include")
+            tools.rename("old_include", os.path.join("include", "aeron"))
+
+            for dll in glob.glob(os.path.join("lib", "*.dll")):
+                shutil.move(dll, "bin")
```
Comment (lukaszlaszko) [inline recipes/aeron/all/conanfile.py:96]: Any reason for moving those from lib to bin?

Comment (uilianries) [reply recipes/aeron/all/conanfile.py:96]: It's the default directory for DLL and executables. All Conan package follow it. It's a MS standard.

File: recipes/aeron/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/aeron/all/conanfile.py:128]: Nope. https://github.com/real-logic/aeron/blob/e45664e171882e34759fa26d0499465732001bb0/aeron-client/src/main/c/CMakeLists.txt#L21

**Pull request merged**